src/test/java/microsoft/exchange/webservices/data/RecurringAppointmentDateParsingTest.java

```diff
@@ -0,0 +1,114 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.TimeZone;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class RecurringAppointmentDateParsingTest extends BaseTest {
+
+	/**
+	 * Parsing the api recurrence date-String, comparing old and new parsing
+	 * code
+	 * @throws ParseException
+	 */
+	@Test
+	public void parseAPIRecurrenceDateFormat() throws ParseException {
+		String dateString = "2015-01-08Z";
+		Date newParsing = exchangeServiceBaseMock.convertStartDateToUnspecifiedDateTime(dateString);
+		Date oldParsing = convertStartDateToUnspecifiedDateTime(dateString);
+		assertEquals(oldParsing, newParsing);
+	}
+
+	/**
+	 * Parsing a previously unparsable recurrence date using the new parser.
+	 * Timezone -1
+	 */
+	@Test
+	public void parsePreviouslyUnbarsableRecurrenceDateFormat() {
+		String dateString = "2015-01-08-01:00";
+		Date parsed = exchangeServiceBaseMock.convertStartDateToUnspecifiedDateTime(dateString);
+		assertNotNull(parsed);
+		Calendar calendar = new GregorianCalendar(TimeZone.getTimeZone("Europe/Paris"));
+		calendar.setTime(parsed);
+		assertEquals(2015, calendar.get(Calendar.YEAR));
+		assertEquals(0, calendar.get(Calendar.MONTH));
+		assertEquals(8, calendar.get(Calendar.DATE));
+		assertEquals(2, calendar.get(Calendar.HOUR));
+		assertEquals(0, calendar.get(Calendar.MINUTE));
+	}
+
+	/**
+	 * Verifying different timezone parsed with new recurrence parser. 
+	 * Timezone +1
+	 */
+	@Test
+	public void parsePreviouslyUnbarsableRecurrenceDateFormat2() {
+		String dateString = "2015-01-08+01:00";
+		Date parsed = exchangeServiceBaseMock.convertStartDateToUnspecifiedDateTime(dateString);
+		assertNotNull(parsed);
+		Calendar calendar = new GregorianCalendar(TimeZone.getTimeZone("Europe/Paris"));
+		calendar.setTime(parsed);
+		assertEquals(2015, calendar.get(Calendar.YEAR));
+		assertEquals(0, calendar.get(Calendar.MONTH));
+		assertEquals(8, calendar.get(Calendar.DATE));
+		assertEquals(0, calendar.get(Calendar.HOUR));
+		assertEquals(0, calendar.get(Calendar.MINUTE));
+	}
+
+	/**
+	 * Parsing timezone -1 using "unmodified" utc parser
+	 */
+	@Test
+	public void parseAsUTC() {
+		String dateString = "2015-01-08-01:00";
+		Date parsed = exchangeServiceBaseMock.convertUniversalDateTimeStringToDate(dateString);
+		assertNotNull(parsed);
+		Calendar calendar = new GregorianCalendar(TimeZone.getTimeZone("Europe/Paris"));
+		calendar.setTime(parsed);
+		assertEquals(2015, calendar.get(Calendar.YEAR));
+		assertEquals(0, calendar.get(Calendar.MONTH));
+		assertEquals(8, calendar.get(Calendar.DATE));
+		assertEquals(2, calendar.get(Calendar.HOUR));
+		assertEquals(0, calendar.get(Calendar.MINUTE));

```

Comment:
The code works fine. I have tested it on real data as well, and the parsed results match the outlook calendar for all the different timezone-formats used in our code.